parquet-hadoop/src/main/java/org/apache/parquet/hadoop/InternalParquetRecordReader.java

```diff
@@ -257,4 +259,64 @@ public boolean nextKeyValue() throws IOException, InterruptedException {
     return Collections.unmodifiableMap(setMultiMap);
   }
 
+  //TODO maintain unmaterializableRecordCounter properly
+  public boolean nextBatch(ObjectColumnVector<T> vector) throws IOException {
```

Comment:
do we need this one? It seems that it is the same as using the regular record reader in a tight loop.